src/Microsoft.IdentityModel.Abstractions/EventDetails.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.IdentityModel.Abstractions
+{
+    /// <summary>
+    /// Details of the telemetry event.
+    /// </summary>
+    public abstract class EventDetails
```

Comment:
It would help to have a console app or a unit test that shows an implementation of system.

Reply:
Not sure I follow the Q. It's used to capture the details of a telemetry event. It's an extensibility point to hopefully capture any embellishments that are needed for events going forward. E.g. if we decide for some reason that public certs should be able to be logged in the telemetry pipe, EventDetails could be extended and a default behavior could be added in that case. Does that answer your Q?

Reply:
@TimHannMSFT, I think some unit tests should help understand the details.

Reply:
@TimHannMSFT that was exactly the answer I was looking for.
If i had phrased the question" "Do you have an example of how to use this class", that would have expressed my intent better.

Reply:
Unit tests for this class are probably worthwhile. I'll add those.